Library/Formula/llvm.rb

```diff
@@ -198,12 +196,6 @@ def install
 
     args << "-DLLVM_ENABLE_RTTI=On" if build.with? "rtti"
 
-    if build.with? "assertions"
-      args << "-DLLVM_ENABLE_ASSERTIONS=On"
-    else
-      args << "-DCMAKE_CXX_FLAGS_RELEASE='-DNDEBUG'"
```

Comment:
That flag was added fairly recently (see #41669) so my assumption was that it is still needed. So yes, in release mode it is building without assertions by default, but additionally passing that argument seems to be still necessary, too. (Note that I haven't checked that myself.)